### PR TITLE
Overload IOException constructor

### LINKED ISSUE #271: Add IOException constructor(message: Throwable?)
`IOException` lacks the secondary constructor `public constructor(message: Throwable?)`. The standard exceptions in the Kotlin stdlib and `java.io.IOException` (= the JVM actual) all have this constructor.

Would it be possible to add it?

Background: this would let us mechanically (without special casing this constructor) replace java.io.IOException with kotlinx.io.IOException as we migrate our codebase to KMP.

### PR DESCRIPTION
**Pull request by Goooler**
Closes #271.

### CODE REVIEW

File: core/native/src/-NonJvmPlatform.kt [hunk 1]
```diff
@@ -30,7 +30,9 @@ public actual open class IOException actual constructor(
 ) : Exception(message, cause) {
     public actual constructor(message: String?) : this(message, null)
 
-    public actual constructor() : this(null)
+    public actual constructor(cause: Throwable?) : this(null, cause)
```
Comment (martinkretzschmar) [inline core/native/src/-NonJvmPlatform.kt:33]: The usual behavior of this exception constructor is `this(cause?.toString(), cause)`.

File: core/common/src/-CommonPlatform.kt [hunk 2]
```diff
@@ -29,6 +29,7 @@ internal expect fun String.asUtf8ToByteArray(): ByteArray
 public expect open class IOException(message: String?, cause: Throwable?) : Exception {
     public constructor()
     public constructor(message: String?)
+    public constructor(cause: Throwable?)
```
Comment (fzhinkin) [inline core/common/src/-CommonPlatform.kt:32]: It does make sense to also add a constructor accepting both a message and a cause.

**Comment by fzhinkin** [comment]
@Goooler thanks for opening the PR!
Please base the PR agains the `develop` branch, not the `master` (as per https://github.com/Kotlin/kotlinx-io/blob/master/CONTRIBUTING.md#submitting-changes).

**Review by fzhinkin — CHANGES_REQUESTED** [review]

File: core/native/src/-NonJvmPlatform.kt (hunk 1 above)
Comment (fzhinkin) [reply core/native/src/-NonJvmPlatform.kt:33]: I'd say it should be super(cause), and all other ctors should delegate to super ctors too.

Like here https://github.com/JetBrains/kotlin/blob/master/libraries/stdlib/wasm/src/kotlin/Exceptions.kt

File: core/common/src/-CommonPlatform.kt (hunk 2 above)
Comment (Goooler) [reply core/common/src/-CommonPlatform.kt:32]: It's already handled by the primary constructor.

File: core/native/src/-NonJvmPlatform.kt (hunk 1 above)
Comment (Goooler) [reply core/native/src/-NonJvmPlatform.kt:33]: I'll send a new PR after this gets merged.

**Review by fzhinkin — APPROVED** [review]

File: core/native/src/-NonJvmPlatform.kt (hunk 1 above)
Comment (Goooler) [reply core/native/src/-NonJvmPlatform.kt:33]: It's in #281.

**Comment by martinkretzschmar** [comment]
Thank you @Goooler!

**Pull request merged**